packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -359,6 +359,20 @@ class _TextLayout {
     final bool hasTrailingSpaces = switch (rawString.codeUnitAt(rawString.length - 1)) {
       0x9 ||        // horizontal tab
       0x3000 ||     // ideographic space
+      0x1680 ||     // ogham space mark
+      0x2000 ||     // en quad
+      0x2001 ||     // em quad
+      0x2002 ||     // en space
+      0x2003 ||     // em space
+      0x2004 ||     // three-per-em space
+      0x2005 ||     // four-per-em space
+      0x2006 ||     // six-per-em space
+      0x2007 ||     // figure space
+      0x2008 ||     // punctuation space
+      0x2009 ||     // thin space
+      0x200A ||     // hair space
+      0x202F ||     // narrow no-break space(NNBSP)
+      0x205F ||     // medium mathematical space(MMSP)
```

Comment:
FTR, this is how SkParagraph determines whether if a cluster is a space (or break). 
https://github.com/google/skia/blob/8b2f189cfa59d0c69b700bbe8587f52f5bd2ad5c/modules/skparagraph/src/ParagraphImpl.cpp#L357-L381

With the ICU backend, `::kPartOfWhiteSpaceBreak` is
https://github.com/unicode-org/icu/blob/23d9628f88a2d0127c564ad98297061c36d3ce77/icu4c/source/common/unicode/uchar.h#L3388-L3425

Reply:
Also you could use regex for matching unicode properties, e.g.: `RegExp(r'\p{Space_Separator}', unicode: true)`